pymc/model.py

```diff
@@ -920,111 +927,112 @@ def test_point(self) -> Dict[str, np.ndarray]:
     @property
     def initial_point(self) -> Dict[str, np.ndarray]:
         """Maps free variable names to transformed, numeric initial values."""
-        if set(self._initial_point_cache) != {get_var_name(k) for k in self.initial_values}:
-            return self.recompute_initial_point()
-        return self._initial_point_cache
+        seed = self.rng_seeder.randint(2 ** 30, dtype=np.int64)
+        return self.recompute_initial_point(seed)
 
-    def recompute_initial_point(self) -> Dict[str, np.ndarray]:
-        """Recomputes numeric initial values for all free model variables.
+    def recompute_initial_point(self, seed) -> Dict[str, np.ndarray]:
+        """Recomputes the initial point of the model.
 
         Returns
         -------
-        initial_point : dict
-            Maps free variable names to transformed, numeric initial values.
-        """
-        self._initial_point_cache = Point(list(self.initial_values.items()), model=self)
-        return self._initial_point_cache
-
-    @property
-    def initial_values(self) -> Dict[TensorVariable, np.ndarray]:
-        """Maps transformed variables to initial values.
-
-        ⚠ The keys are NOT the objects returned by, `pm.Normal(...)`.
-        For a name-based dictionary use the `initial_point` property.
+        ip : dict
+            Maps names of transformed variables to numeric initial values in the transformed space.
         """
-        return self._initial_values
+        fn = self.make_initial_point_fn(return_transformed=True)
+        return Point(fn(seed), model=self)
 
-    def set_initval(self, rv_var, initval):
-        if initval is not None:
-            initval = rv_var.type.filter(initval)
-
-        test_value = getattr(rv_var.tag, "test_value", None)
-
-        rv_value_var = self.rvs_to_values[rv_var]
-        transform = getattr(rv_value_var.tag, "transform", None)
-
-        if initval is None or transform:
-            initval = self._eval_initval(rv_var, initval, test_value, transform)
-
-        self.initial_values[rv_value_var] = initval
-
-    def _eval_initval(
+    def make_initial_point_fn(
```

Comment:
Should this be merged with the make_expression function?

Reply:
Both functions are rather complex on their own already and the split gives us a more fine-grained test surface.

I don't want to rule out further refactoring in follow-ups though. In many ways this PR is just the first step into a new era of initvals and I find it hard to predict what changes we'll make when refactoring `init_nuts`.